File: 199-binary-tree-right-side-view/binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        def dfs(root,depth):
            if(root==None):
                return None
            if(depth==len(res)):
                res.append(root.val)
            dfs(root.right,depth+1)
            dfs(root.left,depth+1)
        res=[]
        dfs(root,0)
        return res



        '''
        if root==None:
            return []

        q=deque([root])
        res=[]

        while q:
            level=[]
            for _ in range(len(q)):
                node=q.popleft()
                level.append(node.val)

                if node.left!=None:
                    q.append(node.left)
                if node.right!=None:
                    q.append(node.right)
            res.append(level[-1])
        return res'''

File: 199-binary-tree-right-side-view/binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public void dfs(List<Integer>  res,TreeNode root,int depth){
        if(root==null){
            return;
        }
        if(res.size()==depth){
           res.add(root.val);
        }
        dfs(res,root.right,depth+1);
        dfs(res,root.left,depth+1);
    }
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer>  res=new ArrayList<>();
        dfs(res,root,0);
        return res;
    }
}